File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.go
func longestPalindrome(s string) string {
    res := ""
    resLen := 0

    for i:= 0; i < len(s); i++{
        // odd case
        l, r := i, i
        for l >= 0 && r < len(s) && s[l] == s[r] {
            if (r -l + 1) > resLen{
                resLen = r - l + 1
                res = s[l: r + 1]
            }
            l--
            r++
        }

        // even case
        l, r = i, i + 1 
        for l >= 0 && r < len(s) && s[l] == s[r] {
            if (r - l + 1) > resLen{
                resLen = r - l + 1
                res = s[l: r + 1]
            }
            l--
            r++
        }

    }
    return res
    
}

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        def expand(l, r):
            while l >= 0 and r < len(s) and s[l] == s[r]:
                l -= 1
                r += 1
            return s[l + 1: r]
        
        res = ""
        for i in range(len(s)):
            # odd case
            odd_palindrome = expand(i, i)
            if len(odd_palindrome) > len(res):
                res = odd_palindrome
            # even case
            even_palindrome = expand(i, i +1)
            if len(even_palindrome) > len(res):
                res = even_palindrome
        return res

        